src/components/Plugin/ContributionRewardExtRewarders/DetailsPageRouter.tsx

```diff
@@ -30,7 +30,10 @@ class DetailsPageRouter extends React.Component<IProps, IStateProps>
 
   public async componentDidMount() {
     if (!this.state.crxDetailsComponent) {
-      this.setState({ crxDetailsComponent: await getCrxRewarderComponent(this.props.data.scheme, CrxRewarderComponentType.Details) });
+      this.setState({ crxDetailsComponent: await getCrxRewarderComponent(
+        (await this.props.data.plugin.entity.fetchState()) as IContributionRewardExtState,
```

Comment:
why is `this.props.data.plugin.entity.fetchState()` not in `withSubscription.createObservable`?

Reply:
I'm unsure how you `.then(...)` chain an observable, can you advise?

Reply:
`.pipe(mergeMap)`?  I'm not sure what you're trying to do.

Is this it:

```
competition.suggestions({}, { subscribe, fetchAllData: true })
    .pipe(
      mergeMap(submissions => of(submissions).pipe(
        mergeMap(submissions => submissions),
        mergeMap((submission: CompetitionSuggestion) => submission.state().pipe(first())),
        toArray()
      )));
```

If so, this was one of my hopes that arc.js 2.0 would be able to provide sugar for.

Reply:
Nope, the idea is that we need to get the plugin entity's state from the proposal entity by fetching it directly. This is because we can't be sure the proposal itemMap initialized the nested entity's state, so we can't just do: `this.props.data.plugin.entity.coreState`

Reply:
I think this should stay as is :P